### PR TITLE
Issue #3180228 by tBKoT: Split permissions for private messages

### PR DESCRIPTION
**Pull request by dmytriikaiun**
## Problem
Disable messaging being initiated by participants to other participants. Also, consider consultants to participants in other programs or geographic areas

## Solution
Extend the current 'use private messaging system' permission with 3:
- create private messages thread
- reply to private messages thread
- delete private messages thread

if the 'use private messaging system' is not selected, 3 following items should not be available
if the 'create private messages' thread is not selected - replying should not be available
hide message icon for users that don't have the 'use private messaging system' permission
update permissions for existing roles:
if a role has permission 'use private messaging system', enable all 3 permission

## Issue tracker
https://www.drupal.org/project/social/issues/3180228
https://getopensocial.atlassian.net/browse/YANG-3886

## How to test
- [x] Edit permissions as you want
- [x] Try to create/reply/delete private messages.

## Screenshots
N/A

## Release notes
Creates 3 new permissions that allow us to configure which roles have access to create/reply/delete private messages.

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Review by ribel — APPROVED** [review]

File: modules/social_features/social_private_message/social_private_message.module [hunk 1]
```diff
@@ -618,3 +636,26 @@ function social_private_message_views_query_alter(ViewExecutable $view, QueryPlu
...
+    ) {
+      return AccessResult::allowed();
+    }
+
+    return AccessResult::forbidden();
+  }
+}
+
+/**
+ * Implements hook_private_message_access().
```
Comment (denysyukm) [inline modules/social_features/social_private_message/social_private_message.module:657]: @dmytriikaiun , could you change "Implements hook_private_message_access()" to "Implements hook_private_message_create_access()", please

**Comment by ribel** [comment]
@ronaldtebrake Since we extend private message permissions with this PR - could you check if all is fine with you and if we can include it to 9.x milestone since we keep BC here.

**Comment by ronaldtebrake** [comment]
@ribel if you can answer the above and it's all fine we can include both in 9.x :) !

File: modules/social_features/social_group/modules/social_group_welcome_message/social_group_welcome_message.module [hunk 2]
```diff
@@ -129,6 +129,14 @@ function social_group_welcome_message_group_content_insert(GroupContentInterface
       $owner = $group->getOwner();
     }
 
+    // Check if user have permission to create private message.
```
[the remark this answers is not in the document]
Comment (dmytriikaiun) [reply modules/social_features/social_group/modules/social_group_welcome_message/social_group_welcome_message.module:132]: @ronaldtebrake yes, you are right. If a group owner doesn't have permission to send PM the member will not receive any notification. I think we can deny access to edit/add a welcome message if a group owner doesn't have permission to create it. However, we need to do it for every user who can edit the group, not only for the owner.

What do you think @ribel?

**Review by ribel — APPROVED** [review]

**Comment by ribel** [comment]
🍒 picked to 9.x

**Pull request merged**